File: Python/Sorting/QuickSort.py
def quickSort(a, low, high):
    if(low<high):
        pivot = a[low]
        i = low
        j = high
        while(True):
            while(pivot > a[i] and a[i] < high):
                i+=1
            while(pivot < a[j] and a[j] > low):
                j-=1
            if(i > j):
                t = a[j]
                a[j] = a[i]
                a[i] = t
            else: break
        a[low] = a[j]
        a[j] = pivot
        quickSort(a,low,j-1)
        quickSort(a,j+1, high)

a = [32,1,34,56,8,6,4,3,43]
print("Before:")
print(a)
quickSort(a, 0, len(a)-1)
print("After:")
print(a)

File: C Programs/Class/quickSort.c
//quickSort
#include<stdio.h>
#define max 7

void quickSort(int a[], int low, int high){
	int pivot, temp, i, j;
	if(low<high){
		pivot = a[low];
		i = low+1;
		j = high;
		
		while(1){
			while(pivot > a[i] && i<=high)
				i++;
			while(pivot < a[j] && j>low)
				j--;
			
			if(i<j){
				temp = a[i];
				a[i] = a[j];
				a[j] = temp;
			}
			else break;
		}
		a[low] = a[j];
		a[j] = pivot;
		
		quickSort(a,low,j-1);
		quickSort(a,j+1,high);
	}
}


int main(){
	int a[max] = {12,5,34,7,98,32,4};
	int i;
	
	printf("Before:\t");
	for(i=0;i<max;i++){
		printf("%d ", a[i]);
	}
	
	quickSort(a, 0, max-1);
	
	printf("\nAfter:\t");
	for(i=0;i<max;i++){
		printf("%d ", a[i]);
	}
	
	return 0;
}
